Remove mention of Array.group[ToMap]

queengooborg opened the pull request:
This PR is a follow-up to #20497 which fixes the data by ~~replacing the improper `alternative_name` with notes and adding a `version_removed` (as confirmed locally).~~ removing all mention of the former methods available on the Array prototype.

Josh-Cena commented:
Can I know what the proper use of `alternative_name` should be? I believe `Intl.Locale` is the same case (https://github.com/mdn/browser-compat-data/pull/19451) and by the same logic should be updated?

queengooborg commented:
`alternative_name` should be used to indicate the non-standard name that the browser uses for that feature, but it should be assumed that it is the same feature otherwise.  It should also be assumed that it is still available on the same interface.

(To be honest though, it may be better to simply remove this data and ignore the one/two Safari versions that `Array.group*` was implemented?)

Josh-Cena commented:
It would be unfair to ignore features implemented in stable browsers. By "it is still available on the same interface", you mean if the method is still an array method (just renamed), it would be considered appropriate for `alternative_name`? That sounds reasonable.

hamishwillee commented:
The last option is to set this false and have a unique entry back for `Array.prototype.groupToMap`. Thanks for asking the question @Josh-Cena - I was confused about this too.

queengooborg commented:
> By "it is still available on the same interface", you mean if the method is still an array method (just renamed), it would be considered appropriate for `alternative_name`?

That's correct!

> It would be unfair to ignore features implemented in stable browsers.

It's true that they are features that were supported in two stable Safari versions.  However, there's a few factors to consider that suggest it isn't important to document:

- It was only supported in two _minor_ Safari releases, of which have been superseded.
- The feature is non-standard and has been superseded with a new feature.
- We have already removed and replaced the feature statement and the associated content pages.
- If we had kept the feature statement, it would have been removed in two years anyways.

While removing the features does mean that we "ignore" them, maintaining complete history of _every feature_ browsers have ever implemented isn't really within the scope of BCD.  (If it was, we'd still be documenting IE-proprietary features.  😝 )  Given all this, I personally don't think it's too helpful to developers to mention the old features.

Josh-Cena commented:
—Okay, I don't have strong opinions. Since it would be you maintaining it anyway, it would be your call if you want to drop it.

queengooborg commented:
Alright, I've now updated this PR to simply remove the mention of the former features!

Josh-Cena commented:
btw, we have this note on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy:

> In some versions of some browsers, this method was implemented as the method Array.prototype.group(). Due to web compatibility issues, it is now implemented as a static method. Check the [browser compatibility table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy#browser_compatibility) for details.

I'm still reluctant to remove this note though, because as a developer I've been constantly advertised about `groupBy`, and it would be somewhat disorienting if the method suddenly changes to a form I don't recognize.

Elchi3 commented:
I think it is fair to apply our 2 year rule here and only remove the alternative name after 2 years of it being unshipped from all released browsers. (see https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#removal-of-irrelevant-features.)

That would also give MDN docs guidance for when to remove the note that @Josh-Cena posted:

> In some versions of some browsers, this method was implemented as the method Array.prototype.group(). Due to web compatibility issues, it is now implemented as a static method. Check the [browser compatibility table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy#browser_compatibility) for details.

So let's leave it for now and then in 2 years MDN and BCD can remove both the irrelevant data and the note in the docs.

Elchi3 commented:
Closing this as we want to leave the alternative name in for the moment.
